app/src/main/java/com/alphawallet/app/widget/BalanceDisplayWidget.java

```diff
@@ -28,50 +26,39 @@
     private final ChainName chainName;
     private final TokenIcon chainIcon;
 
-    private Token token;
-    private Activity activity;
-
     public BalanceDisplayWidget(Context context, @Nullable AttributeSet attrs)
     {
         super(context, attrs);
-        inflate(context, R.layout.item_balance_display,this);
+        inflate(context, R.layout.item_balance_display, this);
         balance = findViewById(R.id.text_balance);
         newBalance = findViewById(R.id.text_new_balance);
         chainName = findViewById(R.id.chain_name);
-
-        token = null;
-
         chainIcon = findViewById(R.id.chain_icon);
-
     }
 
-    public void setupBalance(Token t, TokensService tokenService, Activity act)
+    public void setupBalance(Token token, TokensService tokenService)
     {
-        token = t;
-        activity = act;
-
         chainName.setChainID(token.tokenInfo.chainId);
         chainName.invertChainID(token.tokenInfo.chainId);
-        chainName.setVisibility(View.VISIBLE);
         chainIcon.bindData(tokenService.getToken(token.tokenInfo.chainId, tokenService.getCurrentAddress()), null);
 
-        balance.setText(activity.getString(R.string.total_cost, token.getStringBalance(), token.getSymbol()));
+        balance.setText(getContext().getString(R.string.total_cost, token.getStringBalance(), token.getSymbol()));
     }
 
-    public void setNewBalanceText(BigDecimal transactionAmount, BigInteger networkFee, BigInteger balanceAfterTransaction, boolean isSendingTransaction)
+    public void setNewBalanceText(Token token, BigDecimal transactionAmount, BigInteger networkFee, BigInteger balanceAfterTransaction, boolean isSendingTransaction)
     {
-        balance.setText(activity.getString(R.string.total_cost, token.getStringBalance(), token.getSymbol()));
 
```

Comment:
Just two small changes to this function:

- The formatting is a bit off now; there's spaces and inconsistent brace formatting.
- Now you've increased the clarity of the function it can be seen that 'isSendingTransaction' is actually not required, because it'll always be true if this function is getting called and if token isn't ethereum.

Note that this widget will get more complex when we add NFT handling, but that isn't happening yet.